src/main/scala/demo/MiniDrawer.scala

```diff
@@ -0,0 +1,244 @@
+        .combineWith(theme.mixins_BaseTheme.toolbar)
+    }
+    .build()(c)
+}
+
+val AppBar = ScalaFnComponent.withChildren[Boolean] { (open, c) =>
+  mui.AppBar.styled
+    .fn((theme, _) =>
+      new CSSObject {
+        zIndex = theme.zIndex_BaseTheme.drawer + 1
+        transition = theme.transitions_BaseTheme.create(
+          js.Array("width", "margin"),
+          new Partial {
+            easing = theme.transitions_BaseTheme.easing.sharp
+            duration = theme.transitions_BaseTheme.duration.leavingScreen
+          }
+        )
+
+        if (open) {
+          marginLeft = drawerWidth
+          width = s"calc(100% - ${drawerWidth}px)"
+          transition = theme.transitions_BaseTheme.create(
+            js.Array("width", "margin"),
+            new Partial {
+              easing = theme.transitions_BaseTheme.easing.sharp
+              duration = theme.transitions_BaseTheme.duration.enteringScreen
+            }
+          )
+        }
+      }
+    )
+    .build()
+    .position("fixed")(c)
+}
+
+// todo: hack to set the css property
+trait CssDrawerPage extends CSSObject {
+  @unused
+  @JSName("& .MuiDrawer-paper")
+  var `MuiDrawer-paper`: js.UndefOr[Any | CSSObject] = js.undefined
+}
+
+val Drawer = ScalaFnComponent.withChildren[Boolean] { (open, c) =>
+  mui.Drawer.styled
+    .fn((theme, _) =>
+      new CssDrawerPage {
+        width = drawerWidth
+        flexShrink = 0
+        whiteSpace = "nowrap"
+        boxSizing = "border-box"
+        `MuiDrawer-paper` =
+          if (open) openedMixin(theme)
+          else closedMixin(theme)
+      }.combineWith(
+        if (open) openedMixin(theme)
+        else closedMixin(theme)
+      )
+    )
+    .build()
+    .variant("permanent")(c)
+}
+
+/** Port of https://mui.com/material-ui/react-drawer/#mini-variant-drawer */
+val MiniDrawer = ScalaFnComponent
+  .withHooks[Unit]
+  .withPropsChildren
+  .useState(true)
+  .render { (_, children, openS: UseState[Boolean]) =>
+    val open: Boolean = openS.value
+    def handleDrawerOpen(@unused any: Any): Callback = openS.setState(true)
+    def handleDrawerClose(@unused any: Any): Callback = openS.setState(false)
+
+    mui.Box.sx(new Css { display = "flex" })(
+      mui.CssBaseline(),
+      AppBar(open)(
+        mui.Toolbar(
+          mui.IconButton.normal
+            .color("inherit")
+            .`aria-label`("open drawer")
+            .onClick(handleDrawerOpen)
+            .edge("start")
+            .sx(new Css {
+              marginRight = 5
+              if (open) display = "none"
+            })(muiIcon.Menu()),
+          mui.Typography.variant("h6").noWrap(true).component("div")("Mini variant drawer")
+        )
+      ),
+      Drawer(open)(
+        DrawerHeader(
+          mui.IconButton.normal
+            .onClick(handleDrawerClose)(
+              muiIcon.ChevronLeft() // todo: how to get hold of the theme? {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
```

Comment:
https://mui.com/material-ui/customization/theming/#accessing-the-theme-in-a-component